x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/assets/deferred_transforms_accordion.tsx

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useMemo, useState } from 'react';
+import type { FunctionComponent, MouseEvent } from 'react';
+
+import {
+  EuiAccordion,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSplitPanel,
+  EuiSpacer,
+  EuiText,
+  EuiHorizontalRule,
+  EuiNotificationBadge,
+  EuiButton,
+  EuiToolTip,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import type { ElasticsearchErrorDetails } from '@kbn/es-errors';
+
+import type { EsAssetReference } from '../../../../../../../../common';
+
+import {
+  sendRequestReauthorizeTransforms,
+  useAuthz,
+  useStartServices,
+} from '../../../../../../../hooks';
+
+import { AssetTitleMap } from '../../../constants';
+
+import type { PackageInfo } from '../../../../../types';
+import { ElasticsearchAssetType } from '../../../../../types';
+
+interface Props {
+  packageInfo: PackageInfo;
+  type: ElasticsearchAssetType.transform;
+  deferredInstallations: EsAssetReference[];
+}
+
+export const getDeferredAssetDescription = (assetType: string, assetCount: number) => {
+  switch (assetType) {
+    case ElasticsearchAssetType.transform:
+      return i18n.translate(
+        'xpack.fleet.epm.packageDetails.assets.deferredTransformInstallationsDescription',
+        {
+          defaultMessage:
+            '{assetCount, plural, one {Transform was installed but requires} other {# transforms were installed but require}} additional permissions to run. You must have the transform_admin built-in role or manage_transform cluster privileges to start operations.',
+          values: { assetCount: assetCount ?? 1 },
+        }
+      );
+    default:
+      return i18n.translate(
+        'xpack.fleet.epm.packageDetails.assets.deferredInstallationsDescription',
+        {
+          defaultMessage: 'Asset requires additional permissions.',
+        }
+      );
+  }
+};
+
+export const DeferredTransformAccordion: FunctionComponent<Props> = ({
+  packageInfo,
+  type,
+  deferredInstallations,
+}) => {
+  const { notifications } = useStartServices();
+  const [isLoading, setIsLoading] = useState(false);
+  const deferredTransforms = useMemo(
+    () =>
+      deferredInstallations.map((i) => ({
+        id: i.id,
+        attributes: {
+          title: i.id,
+          description: i.type,
+        },
+      })),
+    [deferredInstallations]
+  );
+
+  const canReauthorizeTransforms =
+    useAuthz().packagePrivileges?.transform?.actions?.canStartStopTransform?.executePackageAction ??
+    false;
+
+  const [transformsToAuthorize, setTransformsToAuthorize] = useState<Array<{
+    transformId: string;
+  }> | null>(null);
+
+  useEffect(() => {
```

Comment:
nit: do we really need a `useEffect` here can we call an `authorizeTransforms` directly from the click events?

Reply:
Updated here [`4e54dee` (#154665)](https://github.com/elastic/kibana/pull/154665/commits/4e54dee3e6ff90782bd6be32f9fa868ddebc777b)